pkg/webreg/webreg.go

```diff
@@ -401,6 +401,116 @@ const templateDefinitions = `
+builds an operator, a bundle and replaces the operator's pull specification in
+manifests with the version built during the test:
+</p>
+
+<code>ci-operator</code> configuration:
+{{ yamlSyntax (index . "optionalOperatorBundleConfig") }}
+
+<p>
+When configuring a bundle build, two options are available:
+</p>
+
+<ul>
+  <li><code>dockerfile_path</code>: path to the Dockerfile that builds the bundle image, defaulting to <code>bundle.Dockerfile</code></li>
+  <li><code>context_dir</code>: base directory for the bundle image build, defaulting to the root of the source tree</li>
+</ul>
+
+<h3 id="index"><a href="#index">Building an Index</a></h3>
+
+<p>
+If <code>ci-operator</code> is configured to build operator bundles from a
+repository, an index image will automatically be built to package those bundles.
+</p>
+
+<h3 id="tests"><a href="#tests">Running Tests</a></h3>
+
+<p>
+Once <code>ci-operator</code> builds the operator bundle and index, they are
+available to be used as a <code>CatalogSource</code> by OLM for deploying and
+testing the operator. Today, the only style of testing that's supported is an
+end-to-end test, implemented with the <a href="/workflow/optional-operators-cvp-common-aws"><code>optional-operators-cvp-common-aws</code></a>
+workflow. This workflow takes the following steps to set up the test environment:
+</p>
+
+<ul>
+  <li>deploy an ephemeral OpenShift cluster to test against</li>
+  <li>create a <code>Namespace</code> to install into</li>
+  <li>create a <code>OperatorGroup</code> and <code>CatalogSource</code> to configure OLM</li>
+  <li>create a <code>Subscription</code> for the operator under test</li>
+  <li>wait for the operator under test to install and deploy</li>
+</ul>
+
+<p>
+A user-provided test can expect to have <code>${KUBECONFIG}</code> set, with
+administrative privileges, and for the operator under test to be fully deployed
+at the time that the test begins. The following example runs a test in this manner:
+</p>
+
+<code>ci-operator</code> configuration:
+{{ yamlSyntax (index . "optionalOperatorTestConfig") }}
+`
+
+const optionalOperatorBundleConfig = `base_images:
+  ubi:               # imports the UBI base image for building
+    namespace: "ocp"
+    name: "ubi"
+    tag: "8"
+  operand:           # imports the latest operand image
+    namespace: "ocp"
+    name: "operand"
+    tag: "latest"
+operator:
+  bundles: # entries create bundle images from Dockerfiles and an index containing all bundles
+  - dockerfile_path: path/to/Dockerfile # defaults to bundle.Dockerfile
+    context_dir: path/                  # defaults to .
+  substitutions:
+  - pullspec: "quay.io/openshift/operand:1.3"  # this will replace references to the operand with the imported version
+    with: "stable:operand"
+  - pullspec: "quay.io/openshift/operator:1.3" # this will replace references to the operator with the built version
+    with: "pipeline:ci-bundle-0"
```

Comment:
`ci-bundle-0` is the bundle image, not the operator image. This should instead point to the operator image which would be in the `images` section in the config.